### PR TITLE
Base TypeScript

### PR DESCRIPTION
**Pull request by devmount**
## Description of the Change

- Add TypeScript dev dependency
- Move main.js -> main.ts
- Update all configuration files to *.config.ts
- Create tsconfig.json with defaults from [Send](https://github.com/thunderbird/send-suite/blob/main/frontend/tsconfig.json)
- Fix Sentry deprecated function calls
- Fix some markdown linter issues

## Benefits

Foundation for all types and type migrations that are to come.

## Applicable Issues

Starts working on #413

### CODE REVIEW

File: frontend/src/main.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ const apiUrl = `${protocol}://${import.meta.env.VITE_API_URL}${port}`;
 app.provide('apiUrl', apiUrl);
 app.provide('bookingUrl', `${protocol}://${import.meta.env.VITE_BASE_URL}/appointments/all/`);
 
-const loc = localStorage?.getItem('locale') ?? (navigator.language || navigator.userLanguage);
```
Comment (devmount) [inline frontend/src/main.ts:61]: I couldn't find any documentation on a `navigator.userLanguage` property, so I removed it 🤷🏻

Comment (MelissaAutumn) [reply frontend/src/main.ts:61]: Apparently it's IE only 😅

**Review by MelissaAutumn — APPROVED** [review]
Looks like a good base!

File: frontend/src/main.ts (hunk 1 above)
Comment (devmount) [reply frontend/src/main.ts:61]: Haha I see, then I guess it's safe to remove it 😅

**Pull request merged**